v2/tools/generator/internal/codegen/pipeline/add_cross_resource_references.go

```diff
@@ -243,6 +106,92 @@ func makeLegacyReferencePropertyName(existing *astmodel.PropertyDefinition, isSl
 	return referencePropertyName
 }
 
+type ARMIDToGenruntimeReferenceTypeVisitor struct {
+	astmodel.TypeVisitor
+	idFactory astmodel.IdentifierFactory
+}
+
+func (v *ARMIDToGenruntimeReferenceTypeVisitor) transformARMIDToGenruntimeResourceReference(_ *astmodel.TypeVisitor, it *astmodel.PrimitiveType, _ interface{}) (astmodel.Type, error) {
+	if it == astmodel.ARMIDType {
+		return astmodel.ResourceReferenceType, nil
+	}
+
+	return it, nil
+}
+
+func (v *ARMIDToGenruntimeReferenceTypeVisitor) renamePropertiesWithARMIDReferences(this *astmodel.TypeVisitor, it *astmodel.ObjectType, ctx interface{}) (astmodel.Type, error) {
+	// First, we visit this type like normal
+	result, err := astmodel.IdentityVisitOfObjectType(this, it, ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	typeName := ctx.(astmodel.TypeName)
+	ot, ok := result.(*astmodel.ObjectType)
+	if !ok {
+		return nil, errors.Errorf("somehow result for visitObjectType of %s was not objectType, instead %T", typeName, result)
+	}
+
+	// Now, check if any properties have been updated and if they have change their names to match
+	var newProps []*astmodel.PropertyDefinition
+	var errs []error
+	ot.Properties().ForEach(func(prop *astmodel.PropertyDefinition) {
+		origProp, ok := it.Property(prop.PropertyName())
+		if !ok {
+			errs = append(errs, errors.Errorf("somehow property %q on %s was lost", prop.PropertyName(), typeName))
```

Comment:
Do we expect this to happen? If so, perhaps tweak the wording to be less wishy-washy.

```suggestion
			errs = append(errs, errors.Errorf("expected to find property %q on %s but failed", prop.PropertyName(), typeName))
```

Reply:
Dropped the "but failed" as that's somewhat obvious given it's an error. Fixed otherwise.